ethcore/src/evm/interpreter.rs

```diff
@@ -644,10 +640,8 @@ impl Interpreter {
 				let init_off = stack.pop_back();
 				let init_size = stack.pop_back();
 				let return_code = mem.read_slice(init_off, init_size);
-				let gas_left = try!(ext.ret(&gas, &return_code));
-				return Ok(InstructionResult::StopExecutionWithGasLeft(
-					gas_left
-				));
+
+				return Ok(InstructionResult::StopExecutionNeedsReturn(gas, return_code.to_owned()))
```

Comment:
`return_code.to_owned()` leads to a potential vulnerability. It requires 2x more memory to be allocated. There is an attack scenario, where attacker pays a lot of for execution of contract which returns very long code.